### ISSUE TITLE
Laggy video with sip video resolution 720p

### ISSUE THREAD

**Issue by NimbleDev** [Renater/SIPMediaGW#11]
Thank you for your great project, which has allowed us to successfully set up a sip video bridge. 
However, we are currently facing an issue: the video becomes extremely laggy, essentially unusable, when the sip source video resolution is set to 720p. 
Do you have any suggestions to solve this problem?

**Comment by nicotyze** [comment]
Thanks for the feedback :)
A key point to keep in mind is that a SIP media gateway is really CPU consuming (=> auto scaling logic strongly recommended). 
This is the output of `docker stats` and `top` commands for a gateway  (gw1) currently running from our side:

![image](https://github.com/Renater/SIPMediaGW/assets/12622750/0356c31c-33fa-4c30-9299-cb93bdb68e3f)



Moreover, in a cloud environment, you should use VM instances which provide the highest CPU performances . For instance, with [this CSP](https://docs.outscale.com/en/userguide/Instance-Types.html): CPU v5 and performance flag = 1 (highest)

**Comment by nicotyze** [comment]
There are mainly two possible causes for "decode_slice_header error": 

- network issues (errors, limited bandwitdth...)
-  a lack of CPU ressources

I see also efps=10.0/23.8 which means that the gateway encodes/sends a video at only 10fps and receives/decodes the video stream coming form your SIP endpoint at 23.8fps.
Classically, these values should be close to efps=30.0/30.0 => there is a bottleneck somewhere...

From my experiments, low framerate values are in most cases due to lack of CPU ressources. Multithreading support for video encoding/decoding processes is quite limited and having many vCPUs may not be enough if the physicall CPU behind is not powerfull enough or if the [vCPU:pCPU](https://download3.vmware.com/vcat/vmw-vcloud-architecture-toolkit-spv1-webworks/index.html#page/Core%20Platform/Architecting%20a%20vSphere%20Compute%20Platform/Architecting%20a%20vSphere%20Compute%20Platform.1.019.html) ratio is to high (for this kind off application it should be 1:1...). 

Did you try with lower video resolutions ?

**Comment by NimbleDev** [comment]
Yes. If we downgrade the video resoluton, the video qualitiy would be stable. But in our use case, we require at least 720p resolution......
Other than using more powerful physical CPU, is there some config options we can try to improve the video performance?
